test/deployment_test.go

```diff
@@ -377,3 +376,48 @@ func (s *TemplateTest) TestPodSecurityContextTakesPrecendenceOverDeprecatedOptio
 
 	s.Require().Equal(int64(2000), *deployment.Spec.Template.Spec.SecurityContext.RunAsUser)
 }
+
+func (s *TemplateTest) TestCanSetSingleTopologySpreadConstraint() {
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"topologySpreadConstraints[0].maxSkew":           "1",
+			"topologySpreadConstraints[0].topologyKey":       "topology.kubernetes.io/zone",
+			"topologySpreadConstraints[0].whenUnsatisfiable": "DoNotSchedule",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", s.Namespace),
+	}
+
+	output := helm.RenderTemplate(s.T(), options, s.ChartPath, s.Release, []string{"templates/deployment.yaml"})
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	s.Require().Equal(int32(1), deployment.Spec.Template.Spec.TopologySpreadConstraints[0].MaxSkew)
+	s.Require().Equal("topology.kubernetes.io/zone", deployment.Spec.Template.Spec.TopologySpreadConstraints[0].TopologyKey)
+	s.Require().Equal(corev1.UnsatisfiableConstraintAction("DoNotSchedule"), deployment.Spec.Template.Spec.TopologySpreadConstraints[0].WhenUnsatisfiable)
+}
+
+func (s *TemplateTest) TestCanSetMultipleTopologySpreadConstraints() {
```

Comment:
The topology value has some special logic around the labelSelector property of the constraint. Would you mind expanding these tests to include that check? Maybe you can set it for one constraint but not the other within this test, or if you would prefer a separate test, that is fine as well.

Reply:
I have added tests around labelSelector. All seem to pass no problem